### ISSUE TITLE
Implement OpenXR support

### ISSUE THREAD

**Issue by morallo** [Prof-Butts/xwa_ddraw_d3d11#24]
Because why not? 😄 

https://www.khronos.org/openxr/

It's phasing out OpenVR as the official API for SteamVR.
Also, it would allow to support Oculus and WMR runtimes directly without needing SteamVR.

**Comment by morallo** [comment]
Hey @Prof-Butts I am giving this a try, and I could use some hints.

First problem: to initialize the XrInstance, I need a pointer to the D3DDevice, which is a member of DeviceResources.
But the deviceResources instance is only created when the app calls DirectDrawCreate(), and not available at the time dllmain is executed, where other initialization routines are run (for example SteamVR).

Should we implement the initialization of VR resources in DeviceResources::Initialize() instead of dllmain?

**Comment by Prof-Butts** [comment]
Yeah, I think that's the right call. Just protect the initialization with some global variable (like g_bUseOpenXR) and maybe try creating a new branch because I've got the feeling several things might break while we work on this.

**Comment by morallo** [comment]
Yep, already created it :-)
This is more an experiment than anything, for now. I will let you know if I manage to make it work.

**Comment by Prof-Butts** [comment]
I'm confident you'll make it work. So, it's a matter of "when", not "if" ;)

I'm also curious about one thing: do we still have to deal with something
similar to WaitGetPoses() and the associated penalty in performance?

**Comment by morallo** [comment]
Yes, all VR runtimes have the same concept, to minimize latency.
In OpenXR there is xrWaitFrame and xrBeginFrame. I hope that the implementation is more flexible and consistent than in openvr.
Oculus SteamVR support was really weird for this.

**Comment by morallo** [comment]
I just remembered another blocker for OpenXR implementation.
Currently Valves OpenXR runtime (part of SteamVR) only supports 64bits. That means we cannot use it.

https://steamcommunity.com/app/250820/discussions/8/3114772813466935534/

So, if we migrate to OpenXR, only HMDs with 32bit runtimes would work, like Oculus, or anything using VirtualDesktoo VDXR runtime.
